src/IterableCodeExtractor.php

```diff
@@ -0,0 +1,122 @@
+<?php
+
+namespace WP_CLI\I18n;
+
+use Gettext\Translation;
+use Gettext\Translations;
+use RecursiveCallbackFilterIterator;
+use RecursiveDirectoryIterator;
+use RecursiveIteratorIterator;
+use DirectoryIterator;
+use WP_CLI;
+
+trait IterableCodeExtractor {
+
+	private static $dir = '';
+
+	/**
+	 * {@inheritdoc}
```

Comment:
I'd use the docs from the [`ExtractorInterface`](https://github.com/oscarotero/Gettext/blob/cae84aff39a87e07bd6e5cddb5adb720a0ffa357/src/Extractors/ExtractorInterface.php) here as there's nothing to inherit from otherwise.

Plus, we could describe the possible options a bit better. Perhaps the way WordPress core does it: https://make.wordpress.org/core/handbook/best-practices/inline-documentation-standards/php/#1-1-parameters-that-are-arrays